sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -95,6 +95,9 @@ class StreamExecution(
   /** The current eventTime watermark, used to bound the lateness of data that will processed. */
   private var currentEventTimeWatermark: Long = 0
 
+  /** The current batch processing timestamp */
+  private var currentBatchTimestamp: Long = 0
```

Comment:
A single thread will change these values. There doesn't need to be locking AFAIK